src/vmemcache_heap.h

```diff
@@ -49,24 +49,45 @@ extern "C" {
 
 #define HEAP_ENTRY_IS_NULL(he) ((he.ptr) == NULL)
 
-struct heap_entry {
+/* extent structure ('struct heap_entry' without header and footer ) */
+struct extent {
 	void *ptr;
 	size_t size;
 };
 
-VEC(extent_vec, struct heap_entry);
-
 struct heap;
 
 struct heap *vmcache_heap_create(void *addr, size_t size, size_t extent_size);
 void vmcache_heap_destroy(struct heap *heap);
 
-ssize_t vmcache_alloc(struct heap *heap, size_t size, struct extent_vec *vec);
-void vmcache_free(struct heap *heap, struct extent_vec *vec);
+ssize_t vmcache_alloc(struct heap *heap, size_t size, void **last_extent,
+			struct extent *first_extent);
+void vmcache_free(struct heap *heap, void *last_extent);
 
 stat_t vmcache_get_heap_used_size(struct heap *heap);
 stat_t vmcache_get_heap_entries_count(struct heap *heap);
 
+void *vmcache_extent_get_prev(void *ptr);
+size_t vmcache_extent_get_size(void *ptr);
+
+/* unsafe variant - the headers of extents cannot be modified */
+#define EXTENTS_FOREACH(ext, extents) \
```

Comment:
surround macro arguments in () when possible

Reply:
Done